### PR TITLE
zfsUnstable: 2.1.12 → 2.1.13-unstable-2023-08-02

### PR DESCRIPTION
**Pull request by toastal**
https://github.com/openzfs/zfs/commits/zfs-2.1.13-staging

With anecdote from a NixOS user that it’s working
https://github.com/openzfs/zfs/pull/15125

…working for me as well as I built+booted from this commit.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/zfs/unstable.nix
Comment (RaitoBezarius) [inline pkgs/os-specific/linux/zfs/unstable.nix:18]: You are not doing the distinction for aarch64, are you?

Comment (toastal) [reply pkgs/os-specific/linux/zfs/unstable.nix:18]: Is there still one? I don’t know the status of ARM…

**Comment by toastal** [comment]
I didn't write this `if`?

**Comment by RaitoBezarius** [comment]
> I didn't write this `if`?

Yes you didn't but you should keep the semantics :P.

**Comment by toastal** [comment]
Oh, so it was removed in the last commit. I see what y’all’re trying to say now.

**Comment by toastal** [comment]
The update has been pushed a while ago.

**Comment by toastal** [comment]
I’ve been booting successfully from this commit (with rebases) for a week now.

**Comment by toastal** [comment]
This has built & been successfully ran for me from kernels

* 6.4.6
* 6.4.7
* 6.4.8
* 6.4.9

**Comment by magneticflux-** [comment]
Should `zfsUnstable` track the [2.2.0 release candidates](https://github.com/openzfs/zfs/releases/tag/zfs-2.2.0-rc3), or just [the `staging` branch](https://github.com/openzfs/zfs/tree/zfs-2.1.13-staging) of the next patch?

**Comment by toastal** [comment]
@magneticflux- 

https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/zfs/unstable.nix#L20-L22

Historically & according to this comment, zfsUnstable points to the most recent staging. If you’re here because you too would like to keep on the latest stable Linux kernel, I too am not sure what the hold up is as this unstable is _supposed_ to follow latest staging from OpenZFS. On that note: tested on 6.4.11 with success too.

**Comment by RaitoBezarius** [comment]
Basically it's moving all the time, I do not have time to test it and I am
on vacations :).

**Comment by toastal** [comment]
@RaitoBezarius but the last commit for 2.1.13 staging in upstream is @RaitoBezarius ’s own commit. Didn’t you test before it was merged?? 😅

**Comment by RaitoBezarius** [comment]
But I don't understand why the PR is being repushed force everyday then?

**Comment by toastal** [comment]
I'm using this merge request as my NixOS config’s Nixpkgs on two machines, otherwise I couldn’t get around the version lock & will need to downgrade to the LTS kernel. To get the latest packages, I gotta rebase.
-- 
toastal ไข่ดาว | https://toast.al
PGP: 7944 74b7 d236 dab9 c9ef  e7f9 5cce 6f14 66d4 7c9e

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**
